File: data structures and algorithms/java/05_038.java
import java.util.*;
import java.util.stream.*;
import java.io.*;
import java.math.*;

public class Main {
    public static void main(String[] args) {
        // Main method code goes here
    }
    
    public class Graph {
        private Map<Object, List<Object>> graph;
        
        public Graph() {
            this.graph = new HashMap<>();
        }
        
        public void bfs(Object start) {
            // Keep track of who we've seen
            Set<Object> visited = new HashSet<>();
            // People to visit next
            Queue<Object> queue = new LinkedList<>();     
            // Add start node to queue
            queue.add(start); 
            // Mark start as seen
            visited.add(start);
            
            while (!queue.isEmpty()) {
                // Get next person to visit
                Object vertex = queue.poll();              
                // Say hello!
                System.out.print(vertex + " ");            
                
                // Check all their friends
                for (Object neighbor : this.graph.get(vertex)) {  
                    // If we haven't met them yet
                    if (!visited.contains(neighbor)) {            
                        // Mark them as seen
                        visited.add(neighbor);                    
                        // Add to people to visit
                        queue.add(neighbor);                      
                    }
                }
            }
        }
    }
}

File: data structures and algorithms/cs/05_038.cs
using System;
using System.Collections.Generic;

public class MainClass
{
   public static void Main(string[] args)
   {
      // Main method code goes here
   }

   public class Graph
   {
      private Dictionary<object, List<object>> graph;

      public Graph()
      {
         this.graph = new Dictionary<object, List<object>>();
      }

      public void Bfs(object start)
      {
         // Keep track of who we've seen
         HashSet<object> visited = new HashSet<object>();
         // People to visit next
         Queue<object> queue = new Queue<object>();
         // Add start node to queue
         queue.Enqueue(start);
         // Mark start as seen
         visited.Add(start);

         while (queue.Count > 0)
         {
            // Get next person to visit
            object vertex = queue.Dequeue();
            // Say hello!
            Console.Write(vertex + " ");

            // Check all their friends
            foreach (object neighbor in this.graph[vertex])
            {
               // If we haven't met them yet
               if (!visited.Contains(neighbor))
               {
                  // Mark them as seen
                  visited.Add(neighbor);
                  // Add to people to visit
                  queue.Enqueue(neighbor);
               }
            }
         }
      }
   }
}
